aioredis/connection.py

```diff
@@ -93,9 +95,11 @@ def create_connection(address, *, db=None, password=None, ssl=None,
         sock = writer.transport.get_extra_info('socket')
         if sock is not None:
             address = sock.getpeername()
-    conn = RedisConnection(reader, writer, encoding=encoding,
-                           address=address, parser=parser,
-                           loop=loop)
+
+    cls = connection_cls or RedisConnection
```

Comment:
You must check that `connection_cls` is subclass of `aioredis.abc.AbcConnection`.
Please add this check before opening connection otherwise (if check fails) you'd need to close reader/writer.